packages/connections/src/utils/validation-errors.spec.ts

```diff
@@ -0,0 +1,284 @@
+import { expect } from 'chai';
+
+import {
+  validateConnectionInfoErrors,
+  validateConnectionInfoWarnings,
+} from './validation';
+import { ConnectionInfo } from 'mongodb-data-service';
+
+describe('Connection String Validation', function () {
+  it('should detect if password is missing', function () {
```

Comment:
we should also test that this works with `scram-sha` and that is not having an error in case a username is not set and the mechanism is also not set